docs/prompting/decomposition/least_to_most.md

```diff
@@ -1,7 +1,99 @@
 ---
-title: ""
-description: ""
-keywords: ""
+title: "Solve simpler subproblems"
+description: "Least-to-Most is a prompting technique that breaks a complex problem down into a series of increasingly complex subproblems."
 ---
 
-[wip]
+Given a complex problem, how can we encourage an LLM to solve simpler subproblems?
+
+Least-to-Most is a prompting technique that breaks a complex problem down into a series of increasingly complex subproblems.
+
+!!! example "Subproblems Example"
+    **original problem**: Adam is twice as old as Mary. Adam will be 11 in 1 year. How old is Mary?
+    
+    **subproblems**: (1) How old is Adam now? (2) What is half of Adam's current age?
+
+These subproblems are solved sequentially, allowing the answers from earlier (simpler) subproblems to inform the LLM while solving later (more complex) subproblems.
+
+```python
+import instructor
+from pydantic import BaseModel
+from openai import OpenAI
+from typing import Iterable
+
+
+class Subquestion(BaseModel):
+    question: str
+
+
+class Answer(BaseModel):
+    answer: int
+
+
+class SubquestionWithAnswers(BaseModel):
+    question: str
+    answer: int
+
+
+client = instructor.from_openai(OpenAI())
+
+
+def decompose(question):
+    return client.chat.completions.create(
+        model="gpt-4o",
+        response_model=Iterable[Subquestion],
+        messages=[
+            {
+                "role": "user",
+                "content": f"Break this question down into subquestions to solve sequentially: {question}",
+            }
+        ],
+    )
+
+
+def solve(question, solved_questions, original_question):
+    return client.chat.completions.create(
+        model="gpt-4o",
+        response_model=Answer,
+        messages=[
+            {
+                "role": "user",
+                "content": f"""
+                This is the original question we are trying to solve: {original_question}
```

Comment:
lets wrap them in xml so start and end of questions make more sense 

<original_question>
{original_question}
</original_question>